### DISCUSSION TITLE
Integration of PocketBase authentication with other backend services

### DISCUSSION THREAD

**Discussion by JanBeelte** [pocketbase/pocketbase#4701] [Q&A] [answered]
Hi all,

I am considering to use PocketBase in an upcoming project, however I will need some additional backend services (RestAPIs) all of which should rely on the same authentication mechanism.
I can think of two possible solutions:
1. Use PocketBase authentication in all APIs: This would mean I need to call a PocketBase endpoint to check whether the token provided by the client is valid. From my research I could only find the refresh endpoint which generates a new token, so not really what I want. **_Is there an API that I could use from my backend service to verify PocketBase tokens?_**
2. Use a standalone authentication service like ory.sh: This would mean I would need a mechanism to connect PocketBase to a third party auth service like ory.sh and bypass the internal auth. I could not find such a feature. **_Is such a feature available or planned?_**

what is your general advice of using PocketBase in such a scenario, would you consider it a good fit?

Thanks for your help!

**Comment by ganigeorgiev** [comment, marked as answer]
> 1. ... Is there an API that I could use from my backend service to verify PocketBase tokens?

We don't have a dedicated endpoint for tokens verification. You can send an ["Auth refresh" request](https://pocketbase.io/docs/api-records/#auth-refresh) - on invalid or expired token it returns an error, otherwise - returns the latest user data and a new token with refreshed `exp` claim (_the old token is still valid and you can discard the new one if you don't need it_).

> 2. ... Is such a feature available or planned?

You can connect 3rd party auth services only using OAuth2. Ory.sh seems to support OpenID Connect (OIDC) so it should work but note that this doesn't _"bypass the internal auth"_ and a copy of the user data will have to exist in PocketBase.

> what is your general advice of using PocketBase in such a scenario, would you consider it a good fit?

There are not enough details to understand the bigger picture and scale of your project, but if you are not sure and planning to use PocketBase in a complex production critical application - then No (_see also https://github.com/pocketbase/pocketbase/discussions/4355_) and you may be better with other more mature solutions (_Supabase, Nhost, SurrealDB, EdgeDB, etc._) or something custom from scratch.

**Reply by JanBeelte** [reply]
Thank you so much for this quick and super helpful response!
I think I will experiment with solution 1 in some open source project (I really love what you guys are building here)
